### PR TITLE
Drop unused InMemoryPodRepository and use SchedulerCommand interface

### PR DESCRIPTION
**Pull request by kvish**
Summary:
* Use only MesosPodRecordRepository, thus drop InMemoryPodRepository
* Incorporate imperative USI interface

### CODE REVIEW

**Review by kensipe — APPROVED** [review]
lgtm

File: src/main/java/org/jenkinsci/plugins/mesos/MesosApi.java [hunk 1]
```diff
@@ -94,26 +93,25 @@ public MesosApi(
...
 
     logger.info("Starting USI scheduler flow.");
-    updates = runScheduler(SpecsSnapshot.empty(), client, materializer).get();
+    commands = runScheduler(client, materializer).get();
   }
 
   /**
-   * Constructs a queue of {@link SpecUpdated} and passes the specs snapshot as the first item. All
-   * updates are processed by {@link MesosApi#updateState(StateEvent)}.
+   * Constructs a queue of {@link SchedulerCommand} and passes the specs snapshot as the first item.
```
Comment (jeschkies) [inline src/main/java/org/jenkinsci/plugins/mesos/MesosApi.java]:
```suggestion
   * Constructs a queue of {@link SchedulerCommand}.
```

File: src/main/java/org/jenkinsci/plugins/mesos/MesosApi.java [hunk 2]
```diff
@@ -94,26 +93,25 @@ public MesosApi(
...
+   * All commands are processed by {@link MesosApi#updateState(StateEventOrSnapshot)}.
```
Comment (jeschkies) [inline src/main/java/org/jenkinsci/plugins/mesos/MesosApi.java]:
```suggestion
   * All state events are processed by {@link MesosApi#updateState(StateEventOrSnapshot)}.
```

**Review by jeschkies — APPROVED** [review]

**Comment by jeschkies** [comment]
It seems we get `NullPointerExceptions`.

**Pull request merged**